File: java/api/src/main/java/com/automatak/CommandResponse.java
package com.automatak;

import com.automatak.dnp3.CommandResult;
import com.automatak.dnp3.CommandStatus;

/**
 * Aggregate command response type.
 *
 * Check the result before checking the status.
 * Status is only valid if response is RESPONSE_OK
 */
public class CommandResponse {

    public CommandResponse(CommandResult result, CommandStatus status)
    {
        this.result = result;
        this.status = status;
    }

    public CommandResult getResult() {
        return result;
    }

    public CommandStatus getStatus() {
        return status;
    }

    private final CommandResult result;
    private final CommandStatus status;

    @Override
    public String toString()
    {
        if(result == CommandResult.RESPONSE_OK) return "Response: " + status.toString();
        else return "Failure: " + result.toString();
    }
}

File: cpp/opendnp3/src/opendnp3/CommandResponse.cpp
#include <opendnp3/CommandResponse.h>

#include <sstream>

namespace opendnp3
{
	
CommandResponse::CommandResponse(CommandResult aResult, CommandStatus aStatus) : 
	mResult(CommandResult::NO_COMMS), 
	mStatus(aStatus) 
{}

CommandResponse CommandResponse::OK(CommandStatus aStatus)
{
	return CommandResponse(CommandResult::RESPONSE_OK, aStatus); 
}
	
CommandResult CommandResponse::GetResult()
{ return mResult; }

CommandStatus CommandResponse::GetStatus()
{ return mStatus; }

bool CommandResponse::operator==(const CommandResponse& arRHS)
{
	return (mResult == arRHS.mResult) && (mStatus == arRHS.mStatus);
}

std::string CommandResponse::ToString() const
{
	std::ostringstream oss;
	if(mResult == CommandResult::RESPONSE_OK) oss << "Response: " << CommandStatusToString(mStatus);
	else oss << "Failure: " << CommandResultToString(mResult);
	return oss.str();
}

}

/* vim: set ts=4 sw=4: */

